lib/rbac/filterer.rb

```diff
@@ -221,7 +221,9 @@ def search(options = {})
       end
 
       if search_filter && targets && (!exp_attrs || !exp_attrs[:supported_by_sql])
-        rejects     = targets.reject { |obj| matches_search_filters?(obj, search_filter, tz) }
+        rejects = TimeProfile.with_cached_time_profile_ids do
```

Comment:
@NickLaMuro Thanks for the very detailed explanation. It makes sense and your understanding of how the profiles are used in those virtual columns seems correct.